src/ert/run_models/everest_run_model.py

```diff
@@ -703,3 +716,42 @@ def _handle_errors(
             self._fm_errors[error_hash]["ids"].append(fm_id)
             error_id = self._fm_errors[error_hash]["error_id"]
             fm_logger.error(err_msg.format(error_id, ""))
+
+
+class SimulatorCache:
+    EPS = float(np.finfo(np.float32).eps)
+
+    def __init__(self) -> None:
+        self._data: defaultdict[
+            int,
+            list[
+                tuple[
+                    NDArray[np.float64], NDArray[np.float64], NDArray[np.float64] | None
+                ]
+            ],
+        ] = defaultdict(list)
+
+    def add(
+        self,
+        realization_id: int,
```

Comment:
what kind of realization is `realization_id`? Would comment or make more explicit name if possible

Reply:
Agreed: `realization_id` -> `realization_name`

Reply:
In the ERT codebase otherwise it is usually called `real` / `realization` , not saying it is ideal, but maybe we shouldn't add yet another synonym? 😅 Unless this is specific to Everest somehow?

Reply:
Reason I used name is that they are strings, and I also use sometimes indices to realizations. I will rename things differently, making sure indices are names appropriately, and go to realizations for the names.